src/wp-includes/option.php

```diff
@@ -2264,6 +2264,17 @@ function delete_network_option( $network_id, $option ) {
 				'site_id'  => $network_id,
 			)
 		);
+
+		if ( $result ) {
```

Comment:
Sorry, on second thoughts I am wondering if this should be 

```suggestion
		if ( false !== $result ) {
```

If `$wpdb::delete()` returns `0`, it means that the query ran successfully but there was no option to delete. The result is the same, it's a known notoption.

If `$wpdb::delete()` returns `false`, then the query failed so it's not known whether or not the option doesn't exist.

Reply:
If there's a problem with the code here, there's the same problem already in trunk: the notoptions-setting code on [lines 2296–2303](https://github.com/WordPress/wordpress-develop/blob/5a76260c3cd22d4688b91620ad7a00a42300c707/src/wp-includes/option.php#L2296-L2303) is guarded by an `if ( $result )` on [line 2269](https://github.com/WordPress/wordpress-develop/blob/5a76260c3cd22d4688b91620ad7a00a42300c707/src/wp-includes/option.php#L2269).

`delete_option()` then has the opposite problem, it sets `notoptions` even if `$result === false`.

Reply:
> `delete_option()` then has the opposite problem, it sets `notoptions` even if `$result === false`.

Good catch. I'll open the original ticket to note that it needs a little work to account for a database failure.

Are you happy to add some protections here or would you prefer me to do it on a separate PR?

Reply:
I think I'd prefer you fix these issues in a separate PR.